Firestore/core/src/firebase/firestore/local/local_serializer.cc

```diff
@@ -103,10 +107,12 @@ MaybeDocument LocalSerializer::DecodeMaybeDocument(
 
   switch (proto.which_document_type) {
     case firestore_client_MaybeDocument_document_tag:
-      return rpc_serializer_.DecodeDocument(reader, proto.document);
+      return DecodeDocument(reader, proto.document,
+                            proto.has_committed_mutations);
```

Comment:
This is subject to the [same bug](https://jpa.kapsi.fi/nanopb/docs/migration.html#fix-undefined-behavior-related-to-bool-fields) we work around in the [remote serializer](https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/core/src/firebase/firestore/remote/serializer.cc#L395).

We probably need a helper function that encapsulates that fix and use it anytime we read a `bool`-valued field. Note that the `has_` accessors seem unaffected based on discussion in [the fix](https://github.com/nanopb/nanopb/commit/eab98ebbd68489987b8399d37cf5517792353fc7).

Note that upgrading to nanopb 0.3.9.4 is also an option for addressing this, but that needs to be done in google3 before we can depend on it in open source. It's likely easier to add the workaround for now.

We can punt resolving this into a separate PR, which is likely worthwhile because we really should audit `bool` fields in our protos and ensure that we're handling all of them uniformly.

Reply:
Done.